src/platform/plugins/shared/data/common/search/search_source/fetch/get_search_params.ts

```diff
@@ -33,16 +33,13 @@ export function getSearchParamsFromRequest(
 ): ISearchRequestParams {
   const { getConfig } = dependencies;
   const searchParams = { preference: getEsPreference(getConfig) };
-  // eslint-disable-next-line @typescript-eslint/naming-convention
-  const { track_total_hits, ...body } = searchRequest.body;
   const dataView = typeof searchRequest.index !== 'string' ? searchRequest.index : undefined;
   const index = dataView?.title ?? `${searchRequest.index}`;
 
+  // @ts-expect-error elasticsearch@9.0.0 `query` types don't match (it seems like it's been wrong here for a while)
```

Comment:
@lukasolson I've checked this and Alejandro is right. The `searchRequest` have a different signature than the return type, and looks like is passing an array of query down to the actual ES search function because the `fetchSearch$` expect an `ISearchRequestParams` that doesn't have the passed `query` array parameter from searchRequest.
Looks like "it just works" but is definitely a tech debt that needs to be fixed soon